steps/reasons-for-appealing/evidence-upload/EvidenceUpload.js

```diff
@@ -37,97 +45,79 @@ class EvidenceUpload extends AddAnother {
-        logger.info(`req.headers: ${JSON.stringify(req.headers)}`);
-        request.post({
-          url: uploadEvidenceUrl,
-          headers: {
-            'Content-Type': 'multipart/form-data'
-          },
-          formData: {
-            file: {
-              value: fileData,
-              options: {
-                filename: fileName,
-                contentType: part.mime,
-                knownLength: req.headers['content-length']
-              }
-            }
+        incoming.once('file', (field, file) => {
+          if (file.name && file.size) {
+            const pathToFile = `${pt.resolve(__dirname, pathToUploadFolder)}/${file.name}`;
+            fs.rename(file.path, pathToFile);
           }
-        }, (forwardingError, resp, body) => {
-          if (forwardingError) {
-            logger.error('Evidence upload error from the api: ', forwardingError);
-            return next(forwardingError);
+        });
+
+        return incoming.parse(req, (uploadingError, fields, files) => {
+          if (files && files.uploadEv &&
+            !fileTypeWhitelist.find(el => el === files.uploadEv.type)) {
+            req.body = {
+              'item.uploadEv': wrongFileTypeError,
+              'item.link': ''
+            };
+            return next();
           }
-          if (resp.statusCode === HttpStatus.OK) {
-            logger.info('No forwarding error, about to save data');
-            const b = JSON.parse(body);
+          if (uploadingError || !get(files, 'uploadEv.name')) {
+            /* eslint-disable operator-linebreak */
+            if (uploadingError &&
+              uploadingError.message &&
+              uploadingError.message.match(/maxFileSize exceeded/)) {
+              /* eslint-enable operator-linebreak */
+              // cater for the horrible formidable.js error
+              /* eslint-disable no-param-reassign */
+              uploadingError = maxFileSizeExceededError;
+              /* eslint-enable no-param-reassign */
+            }
             req.body = {
-              'item.uploadEv': b.documents[0].originalDocumentName,
-              'item.link': b.documents[0]._links.self.href
+              'item.uploadEv': uploadingError,
+              'item.link': ''
             };
             return next();
           }
-          logger.info('Error streaming documents', resp.statusCode, body);
-          req.body = {
-            'item.uploadEv': wrongFileTypeError
-          };
-          return next();
-        });
 
-        part.on('data', incomingData => {
-          fileData.push(incomingData);
-        });
+          const pathToFile = `${pt
+            .resolve(__dirname, pathToUploadFolder)}/${files.uploadEv.name}`;
 
-        part.on('end', () => {
-          fileData.end();
+          return request.post({
+            url: uploadEvidenceUrl,
+            formData: {
+              file: fs.createReadStream(pathToFile)
```

Comment:
Do we no longer need to include the `filename` and `contentType`?

Reply:
We can, but it works just as well without :-o